backend/LeetWars.Core/LeetWars.Core.BLL/Services/NotificationService.cs

```diff
@@ -0,0 +1,108 @@
+﻿using AutoMapper;
+using LeetWars.Core.BLL.Exceptions;
+using LeetWars.Core.BLL.Interfaces;
+using LeetWars.Core.Common.DTO.Challenge;
+using LeetWars.Core.Common.DTO.Friendship;
+using LeetWars.Core.Common.DTO.Notifications;
+using LeetWars.Core.Common.DTO.User;
+using LeetWars.Core.DAL.Context;
+using LeetWars.Core.DAL.Entities;
+using LeetWars.Core.DAL.Enums;
+using Microsoft.EntityFrameworkCore;
+
+namespace LeetWars.Core.BLL.Services
+{
+    public class NotificationServcie : BaseService, INotificationService
+    {
+        private readonly INotificationSenderService _notificationSenderService;
+        private readonly IUserGetter _userGetter;
+        public NotificationServcie(LeetWarsCoreContext context, IMapper mapper, INotificationSenderService notificationSenderService, IUserGetter userGetter) : base(context, mapper)
+        {
+            _notificationSenderService = notificationSenderService;
+            _userGetter = userGetter;
+        }
+
+        public async Task CreateNotification(NotificationDto newNotification)
+        {
+            var notification = _mapper.Map<Notification>(newNotification);
+
+            await _context.Notifications.AddAsync(notification);
+
+            await _context.SaveChangesAsync();
+
+            if(newNotification.ReceiverId is null)
+            {
+                var allIds = _context.Users.Select(e => e.Id);
+                foreach(var id in allIds)
+                {
+                    _context.UserNotifications.Add(new UserNotification
+                    {
+                        IsRead = false,
+                        ReceiverId = id,
+                        NotificationId = notification.Id
+                    });
+                }
+            }
+            else
+            {
+                _context.UserNotifications.Add(new UserNotification
+                {
+                    IsRead = false,
+                    ReceiverId = int.Parse(newNotification.ReceiverId),
+                    NotificationId = notification.Id
+                });
```

Comment:
and all this if/else can be simplify some thing like
var receiversIds = newNotification.ReceiverId is null
    ? await _context.Users.Select(e => e.Id).ToListAsync()
    : new List<long> { newNotification.ReceiverId };
    
   use Select for creating add notifications and save them all at ones